yt/visualization/_commons.py

```diff
@@ -13,28 +14,39 @@
 SUPPORTED_FORMATS = AGG_FORMATS + [".eps", ".ps", ".pdf", ".svg"]
 
 
-def validate_image_name(filename, suffix: str = ".png") -> str:
+def normalize_extension_string(s: str) -> str:
+    return f".{s.lstrip('.')}"
+
+
+def validate_image_name(filename, suffix: Optional[str] = None) -> str:
     """
     Build a valid image filename with a specified extension (default to png).
     The suffix parameter is ignored if the input filename has a valid extension already.
     Otherwise, suffix is appended to the filename, replacing any existing extension.
     """
-    # sanitizing: normalize leading '.'
-    suffix = f".{suffix.lstrip('.')}"
-
     name, psuffix = os.path.splitext(filename)
     if psuffix in SUPPORTED_FORMATS:
-        if suffix:
+        if suffix is not None:
+            suffix = normalize_extension_string(suffix)
+        if suffix in SUPPORTED_FORMATS and suffix != psuffix:
             mylog.warning(
-                "Ignoring supplied suffix '%s' in favour of '%s'", suffix, psuffix
+                "Received two valid image format '%s' and '%s'. "
```

Comment:
```suggestion
                "Received two valid image formats '%s' and '%s'. "
```

Reply:
Good point, I don't think this always happens as close to the surface at it should.